src/app/public/modules/modal/modal.service.ts

```diff
@@ -32,17 +34,30 @@ export class SkyModalService {
     private dynamicComponentService?: SkyDynamicComponentService
   ) { }
 
-  // Open Overloads
-  public open(component: any, providers?: any[]): SkyModalInstance;
-  public open(component: any, config?: IConfig): SkyModalInstance;
+  /**
+   * @private
+   * Removes the modal host from the DOM.
+   */
+  public dispose(): void {
+    if (SkyModalService.host) {
+      this.dynamicComponentService.removeComponent(SkyModalService.host);
+      SkyModalService.host = undefined;
+    }
+  }
 
-  // Open Method
-  public open(): SkyModalInstance {
+  /**
+   * Opens a modal using the specified component.
+   * @param component Determines the component to render.
+   * Since the component generates dynamically instead of with HTML selectors, consumers must
+   * register it with the `entryComponents` property in the `app-extras.module.ts` file.
+   * For more information, see the
+   * [entry components tutorial](https://developer.blackbaud.com/skyux/learn/get-started/advanced/entry-components).
+   * @param {SkyModalConfigurationInterface} config Populates the modal based on the `SkyModalConfigurationInterface` object.
+   */
+  public open(component: any, config?: SkyModalConfigurationInterface | any[]): SkyModalInstance {
```

Comment:
Forgot about this: the `component` parameter should be optional since we used to have an `open()` method (without parameters).

Reply:
Strange. We can move forward then.